### PR TITLE
Simplify local boot and enable multi node

### LINKED ISSUE #1661: add instructions to start a local testnets with X validators
Description:

Scope:

- Create a variable `N_VALIDATORS` that receives a positive integer to start a local testnet
- Update `README`

### PR DESCRIPTION
**Pull request by sabau**
Signed-off-by: Karoly Albert Szabo <szabo.karoly.a@gmail.com>

Closes #1661 

_Description:_

- reduce side effects: 
  - at configuration level: I don't have to know that informations are being placed in my home folder, everything is local to the project and the network name I've chosen
  - at code level helpers are functional and don't rely on global status (first of a series of this kind)
  - You can import helpers without side effect (and test them lately 🎉)
  - if I clone several times the project I can expect them to run indipendently without iteracting on their mutual configurations (of course not at the same time :) )
- taking out the functions to work with multiple nodes
- on the road to reduce the code amount in the boot process for local node
- enabling build + running with one command for multiple nodes
- reduce hardcoded parameters, instead pass them through functions

❤️ Thank you!

---

- [ ] Added entries in `CHANGELOG.md` with issue # and GitHub username
- [ ] Reviewed `Files changed` in the github PR explorer

### CODE REVIEW

File: tasks/testnet.js [hunk 1]
```diff
@@ -18,6 +20,13 @@ async function main() {
     }
     const TESTNET_NODE_FOLDER = path.join(userHome, `.gaiad-testnet`)
     startLocalNode(TESTNET_NODE_FOLDER)
+    if (process.env[`VALIDATORS`] && process.env[`VALIDATORS`] > 1) {
+      // let's re-t
```
Comment (faboweb) [inline tasks/testnet.js]: ?

**Comment by faboweb** [comment]
Awesome :)

File: tasks/testnet.js (hunk 1 above)
Comment (sabau) [reply tasks/testnet.js]: Ah this should have been stashed sorry.
Basically I planned to use the cli() feature to specify how many nodes you want + done some experiments to add accounts.

I moved here the idea about extra parameters that I was playing experimenting on:
https://github.com/cosmos/voyager/issues/1752

Comment (faboweb) [reply tasks/testnet.js]: :D no problem. I like the proposal.

File: tasks/testnet.js [hunk 2]
```diff
@@ -1,27 +1,53 @@
...
+        chainId: network,
+        password: `1234567890`,
+        overwrite: false,
+        moniker: `local`,
+        keyName: `main-account`
+      },
+      numberNodes
+    )
+    await startNodes(nodes, mainAccountSignInfo, network)
+    fs.copySync(join(nodes[1].home, `config`), cliHomePrefix)
```
Comment (sabau) [inline tasks/testnet.js]: This mimic the behaviour of electron start, I tried hard not to put those three lines here and rely on that hook but I couldn't find a nice way of doing it. Any idea?

Comment (sabau) [reply tasks/testnet.js]: From copySync to WriteFileSync should happen at boot in my mind, instead I'm forcing it to happen at network creation time (I didn't wanted to touch also that process right now, to keep this PR small)

Comment (faboweb) [reply tasks/testnet.js]: boot process will die anyway ;)

File: tasks/build/local/helper.js [hunk 3]
```diff
@@ -0,0 +1,223 @@
...
+const saveVersion = nodeHome => {
+  const versionPath = join(nodeHome, `config`)
+  let versionFilePath = join(versionPath, `gaiaversion.txt`) // nodeHome/config is used to copy created config files from
+  return makeExec(
+    `mkdir -p ${versionPath} && ${nodeBinary} version > ${versionFilePath}`
+  )
+}
+
+// nodes[0] is a placeholder just to be aligned with the enumeration used in gaia
+// TODO: next PR refactor also gaia to simplify numeration
```
Comment (faboweb) [inline tasks/build/local/helper.js]: 👍

**Pull request merged**